torch/csrc/jit/codegen/cuda/ir_utils.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include <torch/csrc/jit/codegen/cuda/type.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+
+namespace ir_utils {
+
+#if 0
+template <
+    typename FilterType,
+    template <typename, typename...> typename Container,
+    typename ElementType>
+Container<FilterType*> filterVals(const Container<ElementType*>& container) {
+  Container<FilterType*> out;
+  for (auto& s : container) {
+    if (s->getValType() == FilterType::type) {
+      out.push_back(s->template as<FilterType>());
+    }
+  }
+  return out;
+}
+#else
+template <typename FilterType, typename InputIt>
+class FilterIterator {
+ public:
+  FilterIterator(InputIt first, InputIt last) : input_it_(first), last_(last) {
+    advance();
+  }
+
+  FilterType* operator*() const {
+    return (*input_it_)->template as<FilterType>();
+  }
+
+  FilterIterator& operator++() {
+    ++input_it_;
+    advance();
```

Comment:
could we use `std::find_if` instead?

Reply:
Thanks for the suggestion. Simplified the implementation of `advance` with `find_if`.